### PR TITLE
Opgops 1751

### PR DESCRIPTION
**Pull request by tim-mcarthur-moj**

### CODE REVIEW

File: library/ecs_service.py [hunk 1]
```diff
@@ -268,12 +272,26 @@ def create_service(self, service_name, cluster_name, task_definition,
...
             cluster=cluster_name,
             service=service_name,
             taskDefinition=task_definition,
             desiredCount=desired_count)
+
+        if deployment_maximum_percent is not None or deployment_minimum_healthy_percent is not None:
+            params['deploymentConfiguration'] = {}
+
+            if deployment_maximum_percent is not None:
+                params['deploymentConfiguration']['maximumPercent'] = deployment_maximum_percent
```
Comment (carinadigital) [inline library/ecs_service.py:289]: You should check these are numeric.

File: library/ecs_service.py [hunk 2]
```diff
@@ -268,12 +272,26 @@ def create_service(self, service_name, cluster_name, task_definition,
...
+                       client_token, role):
+        # Boto doesn't like it if you pass defaults for parameters you aren't actually using
+        params = dict(
             cluster=cluster_name,
             service=service_name,
             taskDefinition=task_definition,
             desiredCount=desired_count)
+
+        if deployment_maximum_percent is not None or deployment_minimum_healthy_percent is not None:
+            params['deploymentConfiguration'] = {}
```
Comment (carinadigital) [inline library/ecs_service.py:286]: Does this logic work?
If you don't set `deployment_maximum_percent` or `deployment_minimum_healthy_percent` should the behaviour be to leave the existing ECS settings as is or reset them to empty values?

File: library/ecs_service.py [hunk 3]
```diff
@@ -313,29 +333,30 @@ def main():
     module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
 
     if not HAS_BOTO:
-        module.fail_json(msg='boto is required.')
```
Comment (carinadigital) [inline library/ecs_service.py:316]: Do we have a standard on 2 or 4 spaces?

File: library/ecs_service.py (hunk 1 above)
Comment (brettminnie) [reply library/ecs_service.py:289]: @carinadigital these normally are lifted directly from ansible extra's so not sure how far we want to go into backporting our changes to that repo

File: roles/ec2-app/tasks/rds.yml [hunk 4]
```diff
@@ -38,7 +38,7 @@
 
 - name: Pause for parameter groups to settle
   pause:
-    minutes: 5
+    minutes: 2
```
Comment (brettminnie) [inline roles/ec2-app/tasks/rds.yml:41]: Should we not allow an override on this with a default of 2 minutes?

File: library/ecs_service.py (hunk 3 above)
Comment (tim-mcarthur-moj) [reply library/ecs_service.py:316]: the stuff in the library folder which is not written by is is not maintained by us.  if we want to we can maintain by pulling in newer versions from the original source

File: roles/ecs-app/tasks/create.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+---
+- set_fact:
+    ecs_clusters: "{{ app_data|rejectattr('ecs_cluster', 'undefined')|map(attribute='ecs_cluster')|list|unique }}"
+
+#- block:
+#    - name: Ensure clusters are available for the apps
```
Comment (brettminnie) [inline roles/ecs-app/tasks/create.yml]: For future use? If not please rip it out

File: roles/ecs-app/tasks/destroy.yml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  ecs_service:
+    state: absent
+    cluster: "{{ item.ecs_cluster }}-{{ target}}"
+    name: "{{ item.name }}-{{ target }}"
+  with_items: "{{ typed_apps|selectattr('type','match','ecs')|list}}"
+  register: ecs_svc_gone
+
+
+#- pause: minutes=5
+
```
Comment (brettminnie) [inline roles/ecs-app/tasks/destroy.yml]: If not needed kill it!

**Review by carinadigital — APPROVED** [review]

File: roles/ecs-cluster/tasks/create.yml [hunk 7]
```diff
@@ -4,3 +4,38 @@
     name: "{{ item }}-{{ target }}"
     state: present
   with_items: "{{ ecs_cluster_list|default([]) }}"
+
+#commented for future use when efs data at rest supports encryption
```
Comment (brettminnie) [inline roles/ecs-cluster/tasks/create.yml:8]: Needed for later?

**Review by brettminnie — COMMENTED** [review]
Lots of many commented out things, I am happy with the functionality as I have tested it, however we could do a clean up unless the comments are required for future work

**Pull request merged**